src/base/preferences.cpp

```diff
@@ -561,6 +561,26 @@ void Preferences::setWebUiHttpsKey(const QByteArray &data)
     setValue("Preferences/WebUI/HTTPS/Key", data);
 }
 
+bool Preferences::isAltWebUiEnabled() const
+{
+    return value("Preferences/WebUI/AlternativeUIEnabled", false).toBool();
+}
+
+void Preferences::setAltWebUiEnabled(bool enabled)
```

Comment:
I think re-authentication should be required when users switched this option.

Reply:
No problem to do it. But what is the exact cause?

Reply:
Besides, I don't know how to solve it client-side. It'll just start to recieve `403 Forbidden` to all its API requests.

Reply:
>No problem to do it. But what is the exact cause?

I was thinking the following example: a user use the built-in webUI and is authenticated, now he switch to alternative UI, should he expect to login again or not?

Reply:
Technically there is no problem to keep current session, IIRC.
From the other side, as I said before, I don't know how to handle session break at client side.

Reply:
>I don't know how to handle session break at client side.

No problem just letting them receive error 403 and let them proceed to login manually.

>I was thinking the following example: a user use the built-in webUI and is authenticated, now he switch to alternative UI, should he expect to login again or not?

Does this scenario really make sense?

Reply:
WebUI can be changed at a time when some session is open. It is quite possible scenario.
So should I drop the current sessions at this point?

Reply:
>So should I drop the current sessions at this point?

I thought again, if one can authenticate with qbt now, he can also do it later, so there is no point dropping the sessionID in this case.
Lets keep the current way for simplicity.